src/index.js

```diff
@@ -307,7 +307,16 @@ function cloneElement(element, props, ...children) {
 		elementProps,
 		element.children || elementProps && elementProps.children
 	);
-	return normalizeVNode(preactCloneElement(node, props, ...children));
+	// Only provide the 3rd argument if needed.
+	// Arguments 3+ overwrite element.children in preactCloneElement
+	let cloneArgs = [node, props];
+	if (children && children.length) {
```

Comment:
You cannot drop the if statement. `children` will always be an array because of the spread operator. If you provide an empty array to preactCloneElement it will result in the original children being overwritten with `[]` (nothing).